lib/HLSL/DxilValidationRaceCondCheck.cpp

```diff
@@ -0,0 +1,85 @@
+//===-- DxilValidationRaceCondCheck.cpp - Dxil race condition check pass ---===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+//
+// \file
+// Check race condition for Dxil Validation.
+//
+//===----------------------------------------------------------------------===//
```

Comment:
Add a comment describing what kind of race conditions we can find. Currently I think it is only top level stores (not under any ifs) that store divergent values.